packages/rOpenScPCA/tests/testthat/test-sweep-clusters.R

```diff
@@ -0,0 +1,159 @@
+suppressPackageStartupMessages(library(SingleCellExperiment))
+
+set.seed(2024)
+sce <- splatter::simpleSimulate(nGenes = 1000, verbose = FALSE) |>
+  scater::logNormCounts() |>
+  scater::runPCA(ncomponents = 10)
+
+test_mat <- reducedDim(sce, "PCA")
+
+srat <- Seurat::CreateSeuratObject(counts = counts(sce), assay = "RNA")
+srat[["pca"]] <- Seurat::CreateDimReducObject(
+  embeddings = test_mat,
+  key = "PC_", # underscore avoids Seurat warning that it's adding an underscore
+  assay = "RNA"
+)
+
+
+test_that("sweep_clusters works as expected with default algorithm & weighting", {
+  sweep_list <- sweep_clusters(
+    sce,
+    nn = c(10, 15),
+    resolution = c(0.5, 1)
+  )
+
+  expect_length(sweep_list, 4)
+
+  sweep_list |>
+    purrr::walk(
+      \(df) {
+        expect_setequal(
+          colnames(df),
+          c("cell_id", "cluster", "algorithm", "weighting", "nn", "resolution")
+        )
+
+        # these tests confirm the defaults went through
+        expect_equal(unique(df$algorithm), "louvain")
+        expect_equal(unique(df$weighting), "jaccard")
+
+        expect_true(
+          all(df$nn == 10) || all(df$nn == 15)
+        )
+        expect_true(
+          all(df$resolution == 0.5) || all(df$resolution == 1)
+        )
+      }
+    )
+})
+
+
+
+test_that("sweep_clusters works as expected with matrix input", {
+  sweep_list <- sweep_clusters(
+    test_mat,
+    nn = c(10, 15)
+  )
+
+
+  expect_length(sweep_list, 2)
+
+  sweep_list |>
+    purrr::walk(
+      \(df) {
+        expect_setequal(
+          colnames(df),
+          c("cell_id", "cluster", "algorithm", "weighting", "nn", "resolution")
+        )
+      }
+    )
+})
```

Comment:
Because everything else is already tested, I really meant it that this could be simple!
```suggestion
  sweep_list <- sweep_clusters(test_mat)

  expect_length(sweep_list, 1)
```

Reply:
ah!